src/runtime/eval.c

```diff
@@ -127,6 +127,60 @@ FFS(bits_t x)
 }
 #endif  /* !defined(FFS) */
 
+#if !defined(POPCOUNT)
+uvalue_t POPCOUNT(uvalue_t x) {
+#if defined(__GNUC__)
+  return __builtin_popcountl(x);
+#elif defined(__clang__) && __has_builtin(__builtin_popcountl)
+  return __builtin_popcountl(x);
+#else
```

Comment:
(This also applies to `CLZ` and `CTZ`.)

The reason that these checks aren't combined to a single one is that not all C compilers support it. Some (e.g. MSVC) don't even allow `&&` in preprocessor conditions, but to avoid that, the default implementation would have to be duplicated. The builtins take `unsigned long`s, which have the same size as a pointer on almost all platforms, the only exception I know of is Windows (but that's handled in `config-windows-64.h` and `config-mingw-64.h`).

The fallback implementations are quite naive, but they're simple and when GCC or Clang are used (which is almost always the case), they aren't used anyway.